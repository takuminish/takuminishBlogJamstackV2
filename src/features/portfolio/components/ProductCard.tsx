import { Product } from "../../../pages";
import Card from "@mui/material/Card";
import CardMedia from "@mui/material/CardMedia";
import CardContent from "@mui/material/CardContent";
import Typography from "@mui/material/Typography";
import React from "react";
import Tag from "../../../shared/components/Tag";
import Box from "@mui/material/Box";

type Props = {
  product: Product;
};

function ProductCard(props: Props) {
  const { product } = props;
  return (
    <Card sx={{ maxWidth: 345 }}>
      <CardMedia
        component="img"
        image={product.productImage.url}
        alt="green iguana"
      />
      <CardContent>
        <Typography gutterBottom variant="h6" component="div">
          {product.name}
        </Typography>
        <div
          dangerouslySetInnerHTML={{
            __html: product.detailedInformation,
          }}
        ></div>
        <Box
          sx={{
            display: "flex",
            justifyContent: "start",
            gap: "5px",
          }}
        >
          {product.useSkills.map((skill) => (
            <Tag key={skill} text={skill} />
          ))}
        </Box>
      </CardContent>
    </Card>
  );
}

export default ProductCard;
