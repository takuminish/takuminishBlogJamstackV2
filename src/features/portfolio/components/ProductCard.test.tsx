import React from "react";
import { render } from "@testing-library/react";
import ProductCard, { Props } from "./ProductCard";

const createDOM = () => {
  const props: Props = {
    product: {
      fieldId: "product",
      name: "product1",
      detailedInformation: "<p>test</p>",
      useSkills: ["skill1", "skill2"],
      productImage: {
        url: "https://images.7d0c9b77edc94b119a187f97716ed763/image.webp",
        height: 544,
        width: 1140,
      },
    },
  };
  return render(<ProductCard product={props.product} />);
};

test("snapshot", () => {
  const { container } = createDOM();

  expect(container).toMatchSnapshot();
});
