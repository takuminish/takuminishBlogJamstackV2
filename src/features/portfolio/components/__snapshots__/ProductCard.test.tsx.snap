// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ri6ub7-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="green iguana"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      src="https://images.7d0c9b77edc94b119a187f97716ed763/image.webp"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
      >
        product1
      </div>
      <div>
        <p>
          test
        </p>
      </div>
      <div
        class="MuiBox-root css-1l50nvy"
      >
        <span
          class="MuiBox-root css-1kcf2o3"
        >
          skill1
        </span>
        <span
          class="MuiBox-root css-1kcf2o3"
        >
          skill2
        </span>
      </div>
    </div>
  </div>
</div>
`;
